### PR TITLE
feat: implement feature switch [WEB-535]

### PR DESCRIPTION
**Pull request by gt2345**
## Description

Add `feature_switch` to master config, and every valid switch in the list is considered “on” for feature switching at front end.

## Test Plan

Add `feature_switch` to master config
Verify that `/master` return with `featureSwitch`

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/hooks/useFeature.test.tsx
Comment (ClaireNeveu) [inline webui/react/src/hooks/useFeature.test.tsx:18]: So that this isn't reliant on the current set of feature switches, you can do a cast here:
```typescript
const feature1 = 'feature1' as ValidFeature
const feature2 = 'feature2' as ValidFeature
```
Then you can test if feature1 is on and feature2 is off.

Comment (gt2345) [reply webui/react/src/hooks/useFeature.test.tsx:18]: Great!

File: proto/src/determined/api/v1/master.proto
Comment (ioga) [inline proto/src/determined/api/v1/master.proto:54]: can we please tag it as `= 12`? otherwise it'll conflict with [this SaaS change](https://github.com/determined-ai/determined-ee/blob/master/proto/src/determined/api/v1/master.proto#L62) we are yet to move to OSS.

Comment (gt2345) [reply proto/src/determined/api/v1/master.proto:54]: No problem

**Review by ClaireNeveu — APPROVED** [review]
This looks good to me. Please wait for approval from @ioga before merging.

**Review by ioga — APPROVED** [review]
lgtm

**Pull request merged**
